plugins/bigquery/dbt/adapters/bigquery/impl.py

```diff
@@ -26,6 +29,82 @@
 
 import time
 import agate
+import re
+
+
+BQ_INTEGER_RANGE_NOT_SUPPORTED = f"""
+BigQuery integer range partitioning is only supported by the
+`partition_by` config, which accepts a dictionary.
+
+See: {dbt.links.BigQueryNewPartitionBy}
+"""
+
+
+@dataclass
+class PartitionConfig(JsonSchemaMixin):
```

Comment:
I am LARPing as a person who knows how to use dataclasses. I'm sure there's some subtlety here.... is this the right way to specify these types?

Reply:
Yup!